sdk/storage/Azure.Storage.Common/src/Shared/ClientsideEncryption/Utility.cs

```diff
@@ -102,95 +102,91 @@ public static byte[] CreateKey(int numBits)
             }
 
             return plaintext;
-
+        }
 
 #pragma warning disable CS1587 // XML comment is not placed on a valid language element
-            /// <summary>
-            /// Returns the content encryption key for blob. First tries to get the key encryption key from KeyResolver,
-            /// then falls back to IKey stored on this EncryptionPolicy. Unwraps the content encryption key with the
-            /// correct key wrapper.
-            /// </summary>
-            /// <param name="encryptionData">The encryption data.</param>
-            /// <param name="keyResolver"></param>
-            /// <param name="potentiallyCachedKeyWrapper"></param>
-            /// <param name="async">Whether to perform asynchronously.</param>
-            /// <returns>Encryption key as a byte array.</returns>
-            static async Task<Memory<byte>> GetContentEncryptionKeyAsync(
+        /// <summary>
+        /// Returns the content encryption key for blob. First tries to get the key encryption key from KeyResolver,
+        /// then falls back to IKey stored on this EncryptionPolicy. Unwraps the content encryption key with the
+        /// correct key wrapper.
+        /// </summary>
+        /// <param name="encryptionData">The encryption data.</param>
+        /// <param name="keyResolver"></param>
+        /// <param name="potentiallyCachedKeyWrapper"></param>
+        /// <param name="async">Whether to perform asynchronously.</param>
+        /// <returns>Encryption key as a byte array.</returns>
+        private static async Task<Memory<byte>> GetContentEncryptionKeyAsync(
```

Comment:
This method being a local function has been causing the issue. `async` in the local function has been hiding the same parameter in the `DecryptInternal`. Looks like it has been confusing the analyzer.